pallets/dapp-staking/src/migration.rs

```diff
@@ -498,7 +503,25 @@ impl<T: Config, W: WeightInfo> SteppedMigration for LazyMigration<T, W> {
                 break;
             }
         }
-        log::debug!(target: LOG_TARGET, "migrated {count:?} entries");
+        log::info!(target: LOG_TARGET, "🚚 iterated {count} entries, migrated {migrated}");
         Ok(cursor)
     }
 }
+
+/// Double the remaining block for next era start
+pub struct AdjustEraMigration<T>(PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for AdjustEraMigration<T> {
+    fn on_runtime_upgrade() -> Weight {
+        log::info!("🚚 migrated to async backing, adjust next era start");
+        ActiveProtocolState::<T>::mutate_exists(|maybe| {
+            if let Some(state) = maybe {
+                let current_block_number =
+                    frame_system::Pallet::<T>::block_number().saturated_into::<u32>();
+                let remaining = state.next_era_start.saturating_sub(current_block_number);
+                state.next_era_start.saturating_accrue(2 * remaining);
```

Comment:
Shouldn't this be just `.saturating_accrue(remaining)` since you're directly modifying the existing value?

E.g.:
* current_block = 100
* next_era_start = 130
* remaining = 30
* next_era_start = next_era_start + 30 = 160